### ISSUE TITLE
Reference to a non-existing client filesystem doesn't break, and makes the action disappear/unusable

### ISSUE THREAD

**Issue by grouville** [dagger/dagger#1885]
# Issue
When referencing a non existing client filesystem, `dagger do` doesn't break and makes the action disappear/unusable

# Repro
```
package main

import (
	"dagger.io/dagger"

	"universe.dagger.io/docker/cli"
)

dagger.#Plan & {
	client: {}

	actions: {
		run: cli.#Run & {
			host: "ssh://root@X.X.X.X"
			ssh: {
				key:        client.filesystem."/Users/home/.ssh/id_rsa".read.contents // inexistent in the DAG, but exists on host
				knownHosts: client.filesystem."/Users/home/.ssh/known_hosts".read.contents // inexistent in the DAG, but exists on host
			}
			command: name: "info"
		}
	}
}
```

> The `filesystem` key is not set, which is the surprising fact here. I don't understand why it doesn't trigger an error

# Root cause
I unfortunately don't know how to isolate the root cause

cc @jlongtine

**Comment by helderco** [comment]
Tracked in https://github.com/dagger/dagger/issues/493 and https://github.com/cue-lang/cue/issues/629

**Comment by helderco** [comment]
Fixed by #2334
